### PR TITLE
Cursor row is not adjusted for skipcol with 'smoothscroll'

### PR DESCRIPTION
**Pull request by luukvbaal**
Problem:    Cursor row calculation does not account for skipcol when
            'diff' is not set.
Solution:   Account for skipcol when it is non-zero.

I noticed the `<C-E>` step in the affected test behaved unexpectedly since https://github.com/vim/vim/commit/0222c2d103ad9298bec4dc8864cd80b4e7559db1. Besides the skipcol reset, we also lost a `redraw_later()` with that patch. It seems this is still necessary.

### CODE REVIEW

**Comment by luukvbaal** [comment]
I added a regression test for https://github.com/vim/vim/issues/12409.

File: src/move.c [hunk 1]
```diff
@@ -893,7 +895,7 @@ curs_rows(win_T *wp)
 	}
 	if (valid
 #ifdef FEAT_DIFF
-		&& (lnum != wp->w_topline || !wp->w_p_diff)
+		&& (lnum != wp->w_topline || wp->w_skipcol == 0)
```
Comment (luukvbaal) [inline src/move.c]: I think we need to execute the `else` branch when skipcol is non-zero, not just when 'diff' is set. This affects two screendumps which currently contained invalid cursor positions, which I updated. This in combination with the added redraw_later() also fixes #12411.

Comment (zeertzjq) [reply src/move.c]: This `#ifdef FEAT_DIFF` should be removed then.

Comment (zeertzjq) [reply src/move.c:898]: And the `else` branch is executed when 'diff' is not set.

Comment (zeertzjq) [reply src/move.c:898]: Oops, I read it wrong.

**Comment by luukvbaal** [comment]
Screendumps smooth_wrap_{2,4} are now actually back to how they were before https://github.com/vim/vim/commit/0222c2d103ad9298bec4dc8864cd80b4e7559db1.
Thus I added a testcase from #12411 that failed even before https://github.com/vim/vim/commit/0222c2d103ad9298bec4dc8864cd80b4e7559db1, but passes now.

File: src/move.c (hunk 1 above)
Comment (luukvbaal) [reply src/move.c]: Hmm no I think we need to check for skipcol besides checking for 'diff' filler lines, not instead of. Updated.

**Pull request closed without merging**